### ISSUE TITLE
Inlined context consumer receives wrong value on rerender from provider that uses state

### ISSUE THREAD

**Issue by coderbyheart** [davidmfoley/isolate-components#21]
Below is a minimal reproduction of an issue I've encountered while testing a Context provider that has an internal state which it makes available to consumers. 

What does not work as expected is that the inlined consuming component gets rendered twice, and on the second render receives the wrong value. Note the fifth `console.log` output from within the consumer.

Maybe I am setting this test up wrong?

```
  console.log
    [Provider] selectedTeam is undefined

      at TeamProvider (src/hooks/stateHookContext.spec.tsx:15:4)

  console.debug
    [Provider] Brutal Bearings was selected

      at setSelectedTeam (src/hooks/stateHookContext.spec.tsx:23:18)

  console.log
    [Provider] selectedTeam is Brutal Bearings

      at TeamProvider (src/hooks/stateHookContext.spec.tsx:15:4)

  console.log
    [Consumer] selectedTeam is Brutal Bearings

      at src/hooks/stateHookContext.spec.tsx:42:6

  console.log
    [Consumer] selectedTeam is undefined

      at src/hooks/stateHookContext.spec.tsx:42:6

 FAIL  src/hooks/stateHookContext.spec.tsx
  ✕ should allow to select a team (7 ms)

  ● should allow to select a team

    expect(received).toEqual(expected) // deep equality

    Expected: "Brutal Bearings"
    Received: ""

      48 |      isolated.inline('*')
      49 |
    > 50 |      expect(isolated.content()).toEqual('Brutal Bearings')
         |                                 ^
      51 | })
      52 |

      at Object.toEqual (src/hooks/stateHookContext.spec.tsx:50:29)


```

```typescript
import { isolateComponent } from 'isolate-react'
import { createContext, FunctionComponent, ReactNode, useState } from 'react'

const TeamContext = createContext<{
	selectedTeam?: string
	setSelectedTeam: (team: string) => void
}>({
	setSelectedTeam: () => undefined,
})

const TeamProvider: FunctionComponent<{
	children: ReactNode
}> = ({ children }) => {
	const [selectedTeam, setSelectedTeam] = useState<string | undefined>()

	console.log('[Provider]', 'selectedTeam is', selectedTeam)

	return (
		<TeamContext.Provider
			value={{
				selectedTeam,
				setSelectedTeam: (team) => {
					console.debug('[Provider]', team, 'was selected')
					setSelectedTeam(team)
				},
			}}
		>
			{children}
		</TeamContext.Provider>
	)
}

it('should allow to select a team', async () => {
	const isolated = isolateComponent(
		<TeamProvider>
			<TeamContext.Consumer>
				{({ selectedTeam, setSelectedTeam }) => {
					if (selectedTeam === undefined) {
						setSelectedTeam('Brutal Bearings')
					}
					console.log('[Consumer]', 'selectedTeam is', selectedTeam)
					return selectedTeam
				}}
			</TeamContext.Consumer>
		</TeamProvider>,
	)

	isolated.inline('*')

	expect(isolated.content()).toEqual('Brutal Bearings')
})

```

**Comment by coderbyheart** [comment]
I can see that this is a bit of an unusual set-up because the `setSelectedTeam` is called immediately on render.

When the consuming component is converted to use an event it works:

```typescript
it('should allow to select a team', async () => {
	const isolated = isolateComponent(
		<TeamProvider>
			<TeamContext.Consumer>
				{({ selectedTeam, setSelectedTeam }) => {
					console.log('[Consumer]', 'selectedTeam is', selectedTeam)
					return (
						<button
							onClick={() => {
								setSelectedTeam('Brutal Bearings')
							}}
						>
							{selectedTeam}
						</button>
					)
				}}
			</TeamContext.Consumer>
		</TeamProvider>,
	)

	isolated.inline('*')

	isolated.findOne('button').props.onClick()

	expect(isolated.findOne('button').content()).toEqual('Brutal Bearings')
})
```

**Comment by davidmfoley** [comment]
Thanks for the detailed report.

**Comment by davidmfoley** [comment]
This will be fixed in the next release (v2.3.0)

**Comment by davidmfoley** [comment]
Fixed in 2.3.0
